src/csvinputfileconn.cc

```diff
@@ -413,11 +410,11 @@ namespace dd
 	}
 
       // shuffle before possible test data selection.
-      shuffle_data(ad);
+      shuffle_data(_csvdata, forbid_shuffle);
```

Comment:
That's a bit weird from the outside, call `shuffle`, but `forbid` it via an option :) We'd better control the call from the outside, no ?